src/lighteval/evaluator.py

```diff
@@ -99,10 +102,6 @@ def evaluate(  # noqa: C901
 
     # ===== unpack results and sort back in order and return control to Task =====
     for task_example_id, prediction_list in example_id_response_dict.items():
-        # ===== Unpack the request =====
-        prediction_list.sort(
-            key=lambda x: x.request_index
-        )  # When we use Loglikelihood for several tokens we have all the options here
```

Comment:
This seems to be unnecessary as the lm responses are reordered back to their original order in lm's methods. I put it in this PR because for example when the responses for a document is like `[LoglikelihoodReturn(index=0), LoglikelihoodReturn(index=1), GreedyUntilReturn(index=0)]` (which occurs because `RequestType.LOGLIKELIHOOD` resides before `RequestType.GREEDY_UNTIL` in `RequestType`), this statement wrongly changes it to `[LoglikelihoodReturn(index=0), GreedyUntilReturn(index=0)], LoglikelihoodReturn(index=1)`. If you think we should keep it, we could add the request type to the sort key.

Reply:
The list sorted here will only contain request of the same type. It is used for loglikelihood requests. For example, for one loglikelihood request with 4 choices we would have 4 different request. We sort them here so that we have the same ordering from the task doc.

Reply:
But by looking at the upper for loop, we find that all responses of of all types associated with a single task and single example go to this list.

Reply:
oh yeah i forgot that we could have both greedy and multichoice returns for one task. 
THis could be an issue when computing results. Did you check that removing the sorting was indeed fixing this issue ?